cpp/src/barretenberg/proof_system/composer/composer_helper_lib.hpp

```diff
@@ -33,9 +33,17 @@ std::shared_ptr<typename Flavor::ProvingKey> initialize_proving_key(
     const size_t subgroup_size =
         circuit_constructor.get_circuit_subgroup_size(total_num_constraints + num_randomized_gates); // next power of 2
 
-    auto crs = crs_factory->get_prover_crs(subgroup_size + 1);
+    std::shared_ptr<typename Flavor::ProvingKey> proving_key;
+    if constexpr (IsHonkFlavor<Flavor>) {
+        proving_key = std::make_shared<typename Flavor::ProvingKey>(subgroup_size, num_public_inputs, composer_type);
+    } else if constexpr (IsPlonkFlavor<Flavor>) {
+        // Plonk proving_key needs crs; used in work_queue to compute commitments via direct calls to pippenger
```

Comment:
wdym by direct calls to pippenger?

Reply:
the plonk refactoring was not obvious to me, i agree to your approachof avoiding refactoring :)